File: readability.py
##Prompt user for text:
text = input("Text: ")
##Initalize counters for letters
letters = 0
##Initalize counters for words
words = 0
##Initalize counters for sentences
sentences = 0

##for each character in the text
for char in text:
    ##if character is an alphabetic character, increment the letter counter
    if char.isalpha():
        letters += 1

    ##If chaeracter is a space, increment the word counter
    if char == " ":
        words += 1
    ##If the character is a space, increment the word counter
    if char == "." or char == "!" or char == "?":
        sentences += 1

##Since words are typically separated by spaces, add 1 to the word counter to account for the last word
words += 1

##Calculate the average number of letters per 100 words.
average_letters_per_100_words = (letters / words) * 100

##Calculate the average number of sentences per 100 words.
average_number_of_sentences = (sentences / words) * 100


##Use the Coleman-Liau index formula to calculate the readability grade level.
index = (
    0.0588 * average_letters_per_100_words - 0.296 * average_number_of_sentences - 15.8
)

grade = round(index)

##Print the correct grade level
if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

## Started this problem set on 10/29 in office hours at Widener Library
## Received help from CA João Henrique Santos on printing the correct grade level (L 41), everything else I did by myself

File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)

{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    // printf("%i\n", letters);
    int words = count_words(text);
    //  printf("%i\n", words);
    int sentences = count_sentences(text);
    // printf("%i\n", sentences);

    // Compute the Coleman-Liau index
    float L = ((float) letters / words) * 100.0;
    float S = ((float) sentences / words) * 100.0;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    // Print the grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    if (index > 1)
    {
        printf("Grade %d\n", index);
    }
}

int count_letters(string text)
{
    // Return the number of letters in text
    // Initialize a counter to 0
    // For each character in the text:
    // If the character is a letter:
    //  Increment the counter
    int lettercounter = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            lettercounter++;
        }
    }
    return lettercounter;
}

int count_words(string text)
{
    // Return the number of words in text
    int wordcounter = 1;
    for (int ii = 0; ii < strlen(text); ii++)
    {
        if (text[ii] == ' ')
        {
            wordcounter++;
        }
    }
    return wordcounter;
}

int count_sentences(string text)
{
    // Return the number of sentences in text
    int sentencecounter = 0;
    for (int iii = 0; iii < strlen(text); iii++)
    {
        if (text[iii] == '.' || text[iii] == '!' || text[iii] == '?')
        {
            sentencecounter++;
        }
    }
    return sentencecounter;
}